--- branch.py.orig	2020-05-06 10:27:17.096835546 -0400
+++ branch.py	2020-05-06 10:27:40.353752508 -0400
@@ -141,7 +141,7 @@
             M = 32
         if ~BO[2]:
             CTR = CTR - 1
-        ctr_ok = BO[2] | (CTR[M:64] != 0) ^ BO[3]
+        ctr_ok = BO[2] | SelectableInt((CTR[M:64] != 0), bits=1) ^ BO[3]
         cond_ok = BO[0] | ~(CR[BI + 32] ^ BO[1])
         if ctr_ok & cond_ok:
             NIA = concat(LR[0:62], SelectableInt(value=0x0, bits=2))
