--- sprset.py.orig	2020-05-06 14:15:08.236887767 -0400
+++ sprset.py	2020-05-06 14:15:51.757083567 -0400
@@ -15,11 +15,11 @@
         if n in [13]:
             see(Book_III_p974)
         elif n in [808, 809, 810, 811]:
-            if eq(length(SPR(n)), 64):
+            if eq(len(SPR(n)), 64):
                 SPR[n] = RS
             else:
                 SPR[n] = RS[32:64]
-        elif eq(length(SPR(n)), 64):
+        elif eq(len(SPR(n)), 64):
             SPR[n] = RS
         else:
             SPR[n] = RS[32:64]
