--- sprset.py.orig	2020-05-20 09:27:57.035248195 -0400
+++ sprset.py	2020-05-28 11:02:47.754983231 -0400
@@ -54,7 +54,7 @@
                 n = i
                 count = count + 1
         if eq(count, 1):
-            CR[4 * n + 32:4 * n + 35 + 1] = RS[4 * n + 32:4 * n + 35 + 1]
+            CR.si[4 * n + 32:4 * n + 35 + 1] = RS[4 * n + 32:4 * n + 35 + 1]
         else:
             CR = undefined
         return (CR,)
@@ -78,7 +78,7 @@
                 count = count + 1
         if eq(count, 1):
             RT = concat(0, repeat=64)
-        RT[4 * n + 32:4 * n + 35 + 1] = CR[4 * n + 32:4 * n + 35 + 1]
+        RT[4 * n + 32:4 * n + 35 + 1] = CR.si[4 * n + 32:4 * n + 35 + 1]
         return (RT, CR,)
 
     @inject()
@@ -88,9 +88,9 @@
 
     @inject()
     def op_setb(self, CR):
-        if eq(CR[4 * BFA + 32], 1):
+        if eq(CR.si[4 * BFA + 32], 1):
             RT = SelectableInt(value=0xffffffffffffffff, bits=256)
-        elif eq(CR[4 * BFA + 33], 1):
+        elif eq(CR.si[4 * BFA + 33], 1):
             RT = SelectableInt(value=0x1, bits=256)
         else:
             RT = SelectableInt(value=0x0, bits=256)
