--- sprset.py.orig	2020-05-16 14:04:00.414533305 -0400
+++ sprset.py	2020-05-16 14:04:05.611261221 -0400
@@ -54,7 +54,7 @@
                 n = i
                 count = count + 1
         if eq(count, 1):
-            CR[4 * n + 32:4 * n + 35 + 1] = RS[4 * n + 32:4 * n + 35 + 1]
+            CR.si[4 * n + 32:4 * n + 35 + 1] = RS[4 * n + 32:4 * n + 35 + 1]
         else:
             CR = undefined
         return (CR,)
